core/templates/pages/release-coordinator-page/release-coordinator-page.component.ts

```diff
@@ -105,6 +141,273 @@ export class ReleaseCoordinatorPageComponent implements OnInit {
+              userGroup.userGroupId,
+              cloneDeep(userGroup)
+            );
+            this.newUserGroupName = '';
+            this.cdr.detectChanges();
+          },
+          errorResponse => {
+            this.statusMessage = `Server error: ${errorResponse}`;
+          }
+        );
+    }
+  }
+
+  onUserGroupUserInputChange(): void {
+    this.userInUserGroupValidationError = '';
+    this.userGroupSaveError = '';
+  }
+
+  onUserGroupInputChange(): void {
+    this.userGroupValidationError = '';
+  }
+
+  isUserGroupUpdated(userGroup: UserGroup): boolean {
+    const original = this.userGroupsBackup.get(userGroup.userGroupId);
+    if (original === undefined) {
+      throw new Error(
+        'Backup not found for user group: ' + userGroup.userGroupName
+      );
+    }
+
+    return !isEqual(userGroup, original);
+  }
+
+  _getAllUserGroupNamesExceptSelf(updatedUserGroup: UserGroup): string[] {
+    let userGroupNames: string[] = [];
+    for (let userGroup of this.userGroupsBackup.values()) {
+      if (userGroup.userGroupId !== updatedUserGroup.userGroupId) {
+        userGroupNames.push(userGroup.userGroupName);
+      }
+    }
+    return userGroupNames;
+  }
+
+  _restoreUserGroupToBackup(userGroup: UserGroup): void {
+    let backup = this.userGroupsBackup.get(userGroup.userGroupId);
+    if (backup) {
+      userGroup.userGroupName = backup.userGroupName;
+      userGroup.userGroupUserUsernames = [...backup.userGroupUserUsernames];
+    }
+    return;
+  }
+
+  updateUserGroup(userGroup: UserGroup): void {
+    const ALPHANUMERIC_REGEX = /^[a-zA-Z0-9 ]+$/;
+
+    if (!this.isUserGroupUpdated(userGroup)) {
+      return;
+    }
+
+    if (userGroup.userGroupName.trim() === '') {
+      this.userGroupSaveError = 'User group name should not be empty.';
+      this._restoreUserGroupToBackup(userGroup);
+      return;
+    }
+
+    if (!ALPHANUMERIC_REGEX.test(userGroup.userGroupName)) {
+      this.userGroupSaveError = '';
+      this.userGroupSaveError =
+        'User group name can only contain alphanumeric characters and spaces.';
+      this._restoreUserGroupToBackup(userGroup);
+      return;
+    }
+
+    if (
+      this._getAllUserGroupNamesExceptSelf(userGroup).includes(
+        userGroup.userGroupName
+      ) === true
+    ) {
+      this.userGroupSaveError = '';
+      this.userGroupSaveError = `User group with name ${userGroup.userGroupName} already exist.`;
+      this._restoreUserGroupToBackup(userGroup);
+      return;
+    }
+
+    if (!this.windowRef.nativeWindow.confirm(this.irreversibleActionMessage)) {
+      return;
+    }
+
+    this.statusMessage = 'Updating UserGroups...';
+
+    this.backendApiService
+      .updateUserGroupAsync(
+        userGroup.userGroupId,
+        userGroup.userGroupName.trim(),
+        userGroup.userGroupUserUsernames
+      )
+      .then(
+        () => {
+          this.statusMessage = `UserGroup ${userGroup.userGroupName} successfully updated.`;
```

Comment:
UserGroup --> User group . Please fix similarly elsewhere for user-facing messages.

Also, put quotes around the user group name anytime you interpolate it within a sentence.

Reply:
Updated